### PR TITLE
Removes XHR polyfill

### PR DESCRIPTION
**Pull request by ChrisAntaki**
Swgjs' supported browsers have Beacon and Fetch APIs
- https://caniuse.com/beacon
- https://caniuse.com/fetch

### CODE REVIEW

File: src/runtime/fetcher.js [hunk 1]
```diff
@@ -33,8 +32,8 @@ export class Fetcher {
 
   /**
    * @param {string} unusedUrl
-   * @param {!../utils/xhr.FetchInitDef} unusedInit
-   * @return {!Promise<!../utils/xhr.FetchResponse>}
+   * @param {!Object} unusedInit
```
Comment (fikus) [inline src/runtime/fetcher.js:35]: Can this be typed as `RequestInit`?

File: src/utils/xhr.js [hunk 2]
```diff
@@ -1,369 +0,0 @@
...
-    return this.fetch_(input, init)
-      .catch((reason) => {
-        /*
-         * If the domain is not valid for SwG we return 404 without
-         * CORS headers and the browser throws a CORS error.
-         * We include some helpful text in the message to point the
-         * publisher towards the real problem.
-         */
-        const targetOrigin = parseUrl(input).origin;
-        throw new Error(
```
Comment (fikus) [inline src/utils/xhr.js:104]: Preserve this behavior?

File: src/utils/xhr.js [hunk 3]
```diff
@@ -1,369 +0,0 @@
...
-    const creds = init.credentials;
-    assert(
-      creds === undefined || creds == 'include' || creds == 'omit',
-      'Only credentials=include|omit support: %s',
-      creds
-    );
-    // Fallback to xhr polyfill since `fetch` api does not support
-    // responseType = 'document'. We do this so we don't have to do any parsing
-    // and document construction on the UI thread which would be expensive.
-    if (init.responseType == 'document') {
```
Comment (fikus) [inline src/utils/xhr.js:82]: Do we need to preserve this behavior? responseType in init is not part of the fetch standard AFAICT.

Comment (ChrisAntaki) [reply src/utils/xhr.js:82]: Swgjs only uses `fetch` in a few places, through the Fetcher. In those places, I didn't see any evidence we needed this behavior

File: src/runtime/fetcher.js (hunk 1 above)
Comment (ChrisAntaki) [reply src/runtime/fetcher.js]: That works, added !

File: src/utils/xhr.js (hunk 2 above)
Comment (ChrisAntaki) [reply src/utils/xhr.js:104]: Sure, I just added it here:
suggests devs look at Publisher Center setup when a fetch fails

**Review by fikus — APPROVED** [review]

**Pull request merged**